x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/InternalUsers.java

```diff
@@ -178,7 +178,9 @@ public class InternalUsers {
             UsernamesField.SYNONYMS_ROLE_NAME,
             null,
             new RoleDescriptor.IndicesPrivileges[] {
-                RoleDescriptor.IndicesPrivileges.builder().indices(".synonyms*").privileges("all").allowRestrictedIndices(true).build() },
+                RoleDescriptor.IndicesPrivileges.builder().indices(".synonyms*").privileges("all").allowRestrictedIndices(true).build(),
+                // Need manage privilege to be able to reload index analyzers for all indices
+                RoleDescriptor.IndicesPrivileges.builder().indices("*").privileges("manage").build(), },
```

Comment:
Is it possible to give less permission that `manage`? Can `read`  and `view_index_metadata` permissions work?

Reply:
I checked doing the following setup:

`PUT /_security/role/test-role`
```json
{
  "cluster": [],
  "indices": [
    {
      "names": ["test-index"],
      "privileges": ["read"]
    }
  ],
  "applications": [],
  "run_as": []
}
```

`POST /_security/user/test-user`
```json
{
  "password" : "testtest",
  "roles" : ["test-role"]
}
```

`curl -u test-user:testtest -X POST http://localhost:9200/test-index/_reload_search_analyzers`
```json
{
  "error": {
    "root_cause": [
      {
        "type": "security_exception",
        "reason": "action [indices:admin/reload_analyzers] is unauthorized for user [test-user] with effective roles [test-role] on indices [test-index], this action is granted by the index privileges [manage,all]"
      }
    ],
    "type": "security_exception",
    "reason": "action [indices:admin/reload_analyzers] is unauthorized for user [test-user] with effective roles [test-role] on indices [test-index], this action is granted by the index privileges [manage,all]"
  },
  "status": 403
}
```
I checked using `view_index_metadata`, same result.

Do you see this as a concern that we should raise?

Reply:
Thanks for checking, Carlos. Let's have `manage` role then.

Reply:
have we checked with the security team about permissions?

Reply:
I'm checking with the security team in [this slack conversation](https://elastic.slack.com/archives/CA6N3HDF1/p1689328457266199). I had assumed we were ok for the [reasons stated above](https://github.com/elastic/elasticsearch/pull/97621#issuecomment-1633002289), which might have been wrong. 

Thanks for the ping, Luca!